src/interchange/src/json.rs

```diff
@@ -8,10 +8,203 @@
+        f.debug_struct("JsonEncoder")
+            .field(
+                "schema",
+                &format!("{:?}", build_row_schema_json(&self.value_columns, "schema")),
+            )
+            .finish()
+    }
+}
+
+/// Encodes a sequence of `Datum` as JSON, using supplied column names and types.
+pub fn encode_datums_as_json<'a, I>(
+    datums: I,
+    names_types: &[(ColumnName, ColumnType)],
+) -> serde_json::value::Value
+where
+    I: IntoIterator<Item = Datum<'a>>,
+{
+    let mut name_idx = 0;
+    let value_fields = names_types
+        .iter()
+        .zip_eq(datums)
+        .map(|((name, typ), datum)| {
+            let name = name.as_str().to_owned();
+            (
+                name,
+                TypedDatum::new(datum, typ.clone()).json(&mut move || {
+                    let ret = format!("record{}", name_idx);
+                    name_idx += 1;
+                    ret
+                }),
+            )
+        })
+        .collect();
+    serde_json::value::Value::Object(value_fields)
+}
+
+pub trait ToJson {
+    /// Transforms this value to a JSON value.
+    fn json<F: FnMut() -> String>(self, namer: &mut F) -> serde_json::value::Value;
+}
+
+impl<'a> ToJson for TypedDatum<'_> {
+    fn json<F: FnMut() -> String>(self, namer: &mut F) -> serde_json::value::Value {
+        let TypedDatum { datum, typ } = self;
+        if typ.nullable && datum.is_null() {
+            serde_json::value::Value::Null
+        } else {
+            match &typ.scalar_type {
+                ScalarType::Bool => json!(datum.unwrap_bool()),
+                ScalarType::Int16 => json!(datum.unwrap_int16()),
+                ScalarType::Int32 | ScalarType::Oid => json!(datum.unwrap_int32()),
+                ScalarType::Int64 => json!(datum.unwrap_int64()),
+                ScalarType::Float32 => json!(datum.unwrap_float32()),
+                ScalarType::Float64 => json!(datum.unwrap_int64()),
+                ScalarType::Numeric { .. } => json!(datum.unwrap_numeric().0),
+                // https://stackoverflow.com/questions/10286204/what-is-the-right-json-date-format
+                ScalarType::Date => {
+                    serde_json::value::Value::String(format!("{:?}", datum.unwrap_date()))
+                }
+                ScalarType::Time => {
+                    serde_json::value::Value::String(format!("{:?}", datum.unwrap_time()))
+                }
+                ScalarType::Timestamp => serde_json::value::Value::String(format!(
+                    "{:?}",
+                    datum.unwrap_timestamp().timestamp_millis()
+                )),
+                ScalarType::TimestampTz => serde_json::value::Value::String(format!(
+                    "{:?}",
+                    datum.unwrap_timestamptz().timestamp_millis()
+                )),
+                ScalarType::Interval => {
+                    serde_json::value::Value::String(format!("{:?}", datum.unwrap_interval()))
```

Comment:
This produces ```Interval { months: 12, duration: 0 }``` which is different from what the SQL client will return. Given that there does not seem to be a standard about this, I think we should be returning either the same thing that is being returned by the SQL client or a full-blown JSON fragment that people could actually parse.